src/iOS/toga_iOS/widgets/optioncontainer.py

```diff
@@ -0,0 +1,134 @@
+from rubicon.objc import objc_method
+
+from toga_iOS.libs import (
+    UITabBarController,
+    UITabBarItem,
+    UIViewController,
+)
+
+from toga_iOS.window import iOSViewport
+
+# from .base import Widget
+from toga_iOS.widgets.base import Widget
+
+
+class TogaTabBarController(UITabBarController):
+    """
+    Custom implememtation of the nativie IOS UITabBarController.
+    Used to provide custom delegates:
+        * Tab Bar Selections.
+            - shouldSelect
+            - didSelect
+        * Tab Bar Customizations:
+            - willBeginCustomizing
+            - willEndCustomizing
+            - didEndCustomizing
+        * Rotation Settings:
+            - tabBarControllerSupportedInterfaceOrientations
+            - tabBarControllerPreferredInterfaceOrientationForPresentation
+        * Tab Bar Transition Animations:
+            - animationControllerForTransitionFrom
+            - interactionControllerFor.
+    """
+
+    # @objc_method
+    # def tabBarController_shouldSelectViewController_(self,
+    #                                                  tbc,           # type: UITabBarController
+    #                                                  vc,            # type: UIViewController
+    #                                                  ) -> bool:
+    #     """
+    #     Event callback before tab has transitioned to new tab.
+    #     """
+    #
+    #     # print(f'TogaTabBarController.tabBarController_shouldSelectViewController_(tbc,vc)')
+    #
+    #     result = True
+    #     return result
+
+    @objc_method
+    def tabBarController_didSelectViewController_(self,
+                                                  tbc,              # type: UITabBarController
+                                                  vc,               # type: UIViewController
+                                                  ) -> None:
```

Comment:
Eventually, we're going to adopt Black across all the BeeWare projects; black indentation style is:
```
    def tabBarController_didSelectViewController_(
         self,
         tbc,
         vc,
     ) -> None:
```
rather than the "deep hanging indent" format that PyCharm (and others) prefer.

Also, slightly more verbose names would be preferable - "tabbar_controller" over "tbc"

Reply:
ok